src/track/beats.cpp

```diff
@@ -329,15 +307,65 @@ Bpm BeatsInternal::getBpm() const {
 bool BeatsInternal::isValid() const {
     return m_iSampleRate > 0 && !m_beats.empty();
 }
+void BeatsInternal::setSampleRate(int sampleRate) {
+    m_iSampleRate = sampleRate;
+    updateBpm();
+}
 BeatsInternal::BeatsInternal()
         : m_iSampleRate(0), m_dDurationSeconds(0) {
 }
+
+BeatsInternal::BeatsInternal(const QByteArray& byteArray)
+        : BeatsInternal() {
```

Comment:
I don't think I have ever seen a different constructor of the same class called in the initializer list. Is this necessary?